x-pack/plugins/cloud_integrations/cloud_chat/server/routes/chat.ts

```diff
@@ -42,20 +39,22 @@ export const registerChatRoute = ({
    */
   getChatDisabledThroughExperiments: () => Promise<boolean>;
 }) => {
-  if (!security) {
-    return;
-  }
-
   router.get(
     {
       path: GET_CHAT_USER_DATA_ROUTE_PATH,
       validate: {},
     },
-    async (_context, request, response) => {
-      const user = security.authc.getCurrentUser(request);
-      const { metadata, username } = user || {};
-      let userId = username;
-      let [userEmail] = (metadata as MetaWithSaml)?.saml_email || [];
+    async (context, request, response) => {
+      const { security } = await context.core;
+      const user = security.authc.getCurrentUser();
+
+      if (!user) {
+        // Hide the API from unauthenticated users
+        return response.notFound();
```

Comment:
I wonder if we should instead return `response.unauthorized` in that case, because the API is registered and such route exists, but is available only for authenticated users. So the user is just unauthenticated for the route.
Or this is intentional and I'm missing smth?

Reply:
I agree that unauthorized seems more appropriate. But I didn't want to break the current behavior (if security plugin is not present, it wouldn't register the API, effectively making it a 404).

Happy to iterate if you think it's worth taking the opportunity to correct this.

Reply:
Ah, I see, we can keep it as is then. 👍